tests/main/snap-service-stop-mode-sigkill/task.yaml

```diff
@@ -24,11 +25,11 @@ execute: |
 
     refresh_modes="sigterm sigterm-all"
     for s in $refresh_modes; do
-        systemctl show -p ActiveState snap.test-snapd-service.test-snapd-${s}-service | MATCH "ActiveState=active"
+        systemctl show -p ActiveState "snap.test-snapd-service.test-snapd-${s}-service" | MATCH "ActiveState=active"
     done
 
     echo "we expect two sleep processes (children) from the two sigterm services"
-    n=$(ps afx | grep [3]133731337 | grep -v grep | wc -l)
+    n=$(pgrep -c -f '[3]133731337')
```

Comment:
You don't need the `[3]` character class trick when using `pgrep`, as per pgrep(1), pgrep "will never report itself as a match"

Reply:
Thanks for the tip, will push an update.